lib/segment/src/segment_constructor/segment_builder.rs

```diff
@@ -161,6 +161,14 @@ impl SegmentBuilder {
                                 new_internal_id,
                                 &other_payload_index.payload(old_internal_id)?,
                             )?;
+                            existing_internal_id
+                        } else {
+                            // Old version is still good, do not move anything else
+                            // Mark newly added vector as removed
+                            new_internal_id
+                        };
+                        for vector_storage in vector_storages.values_mut() {
+                            vector_storage.delete_vector(remove_id)?;
```

Comment:
Recovered logic according to old versions: https://github.com/qdrant/qdrant/blob/v1.0.3/lib/segment/src/segment_constructor/segment_builder.rs#L144